### PR TITLE
Evaluation of promises in test code, and phantom-driver cleanup

### PR DESCRIPTION
**Pull request by stefwalter**
This pull requests implements waiting on promises in test code. So we can evaluate an expression that returns a promise, and in python we'll only get back a response (or exception) when the promise completes.

In addition I've cleaned up the phantom-driver and Phantom implementations.

### CODE REVIEW

**Comment by stefwalter** [comment]
This blocks other pull requests, and fixes some test races.

**Comment by petervo** [comment]
Test failures seem legitimate, I think test/storagelib may need some work

File: test/check-multi-machine-key [hunk 1]
```diff
@@ -68,20 +68,7 @@ KEY_IDS = [
...
-                    window.test_result = data;
-                }).
-                fail(function (error) {
-                    console.log(error);
-                });
-        });
-    })""",)
-
-    b.wait_js_func("(function (expected) { return window.test_result == expected; })", "\n".join(keys) + "\n")
+    return b.eval_js("require('base1/cockpit').spawn([ 'ssh-add', '-l' ])")
```
Comment (petervo) [inline test/check-multi-machine-key]: Is this still checking that given keys are present in the agent?

Comment (stefwalter) [inline test/check-multi-machine-key]: Good catch. Fixed.

**Comment by stefwalter** [comment]
Current failures are #2915

**Comment by stefwalter** [comment]
This failure is #2939 

```
Error: #file-content not found
```

**Pull request closed without merging**